### PR TITLE
contrib/net/http: add method RTWithSpanOptions

### PR DESCRIPTION
**Pull request by afarbos**
Allowing to provide additional options
when starting a new span with the round tripper.

Related issue: https://github.com/DataDog/dd-trace-go/issues/984

### CODE REVIEW

File: contrib/net/http/roundtripper_test.go [hunk 1]
```diff
@@ -308,3 +308,25 @@ func TestResourceNamer(t *testing.T) {
...
+
+	const (
+		tagKey   = "foo"
+		tagValue = "bar"
+	)
+	var (
+		mt     = mocktracer.Start()
+		rt     = WrapRoundTripper(http.DefaultTransport, RTWithSpanOptions(tracer.Tag(tagKey, tagValue)))
+		client = &http.Client{Transport: rt}
+	)
```
Comment (knusbaum) [inline contrib/net/http/roundtripper_test.go:324]: Let's just use the `:=` syntax instead of the `const` and `var` blocks.

Comment (knusbaum) [inline contrib/net/http/roundtripper_test.go:324]: Let's just use the `:=` syntax instead of the `const` and `var` blocks.

**Review by knusbaum — CHANGES_REQUESTED** [review]
I just have a style comment.

**Comment by afarbos** [comment]
> @afarbos any interest to continue this? It's been a month.

Sorry for some reason I did not receive the notification.

**Review by knusbaum — APPROVED** [review]
LGTM, Thanks.

**Pull request merged**